pymc3/sampling.py

```diff
@@ -903,7 +903,18 @@ def _iter_sample(
     try:
         step.tune = bool(tune)
         if hasattr(step, 'reset_tuning'):
-            step.reset_tuning()
+            if isinstance(step, CompoundStep):
+                if hasattr(step.methods[0], 'is_mlda_base'):
+                    if not step.methods[0].is_mlda_base:
+                        step.reset_tuning()
+                else:
+                    step.reset_tuning()
+            else:
+                if hasattr(step, 'is_mlda_base'):
+                    if not step.is_mlda_base:
```

Comment:
couldn't this check be implemented in the step methods 'reset_tuning' implementation?

Reply:
Moved the logic within metropolis' reset_tuning()